app/assets/javascripts/discourse/app/components/sidebar/messages-section.js

```diff
@@ -1,10 +1,99 @@
 import { action } from "@ember/object";
+import { cached } from "@glimmer/tracking";
 
 import GlimmerComponent from "discourse/components/glimmer";
 import Composer from "discourse/models/composer";
 import { getOwner } from "discourse-common/lib/get-owner";
+import GroupMessageSectionLink from "discourse/lib/sidebar/messages-section/group-message-section-link";
+import PersonalMessageSectionLink from "discourse/lib/sidebar/messages-section/personal-message-section-link";
+
+export const INBOX = "inbox";
+const UNREAD = "unread";
+const SENT = "sent";
+const NEW = "new";
+const ARCHIVE = "archive";
+
+export const PERSONAL_MESSAGES_INBOXES = [INBOX, UNREAD, NEW, SENT, ARCHIVE];
+export const GROUP_MESSAGES_INBOXES = [INBOX, UNREAD, NEW, ARCHIVE];
 
 export default class SidebarMessagesSection extends GlimmerComponent {
+  constructor() {
+    super(...arguments);
+
+    this.appEvents.on(
+      "page:changed",
+      this,
+      this._refreshSectionLinksDisplayState
+    );
+  }
+
+  willDestroy() {
+    this.appEvents.off(
+      "page:changed",
+      this,
+      this._refreshSectionLinksDisplayState
+    );
+  }
+
+  _refreshSectionLinksDisplayState({
+    currentRouteName,
+    currentRouteParentName,
+    currentRouteParams,
+  }) {
+    const sectionLinks = [
+      ...this.personalMessagesSectionLinks,
+      ...this.groupMessagesSectionLinks,
+    ];
+
+    if (
+      !currentRouteParentName ||
+      currentRouteParentName !== "userPrivateMessages"
+    ) {
```

Comment:
```suggestion
    if (currentRouteParentName !== "userPrivateMessages") {
```
I _think_ it can be simplified to this (man, boolean arithmetic fries my brain)

Reply:
Yes this makes sense haha, the original condition was a mindbender